### PR TITLE
Refs #23919 -- Removed usage of obsolete SafeBytes class.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

**Comment by claudep** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
Seems like that needs a deprecation or a mention as a backwards incompatible change?

**Comment by timgraham** [comment]
It's not clear to me why this should be removed now as opposed to remaining as a stub and included in https://code.djangoproject.com/ticket/27753.

**Comment by claudep** [comment]
OK, I reintroduced a minimal version to not break user code imports, but Django itself should not use it any longer.

File: docs/howto/custom-template-tags.txt [hunk 1]
```diff
@@ -199,11 +199,10 @@ passed around inside the template code:
   They're commonly used for output that contains raw HTML that is intended
   to be interpreted as-is on the client side.
 
-  Internally, these strings are of type ``SafeBytes`` or ``SafeText``.
-  They share a common base class of ``SafeData``, so you can test
-  for them using code like::
+  Internally, these strings are of type ``SafeText``. You can test for them
```
Comment (timgraham) [inline docs/howto/custom-template-tags.txt]: Could you make this a link and add an import for `SafeText` in the code example?

File: django/utils/encoding.py [hunk 2]
```diff
@@ -231,20 +228,18 @@ def filepath_to_uri(path):
...
 
-    Assume the input is either UTF-8 bytes or a string.
-
     This method will encode certain chars that would normally be recognized as
     special chars for URIs.  Note that this method does not encode the '
     character, as it is a valid character within URIs.  See
     encodeURIComponent() JavaScript function for more details.
 
-    Returns an ASCII string containing the encoded result.
+    Returns a string containing the result.
```
Comment (timgraham) [inline django/utils/encoding.py]: Return

File: django/utils/safestring.py [hunk 3]
```diff
@@ -22,6 +22,8 @@ class SafeBytes(bytes, SafeData):
     """
     A bytes subclass that has been specifically marked as "safe" (requires no
     further escaping) for HTML output purposes.
+    Kept in Django 2.0 for backwards-compatibility, but should not be actively
```
Comment (timgraham) [inline django/utils/safestring.py]: suggested wording: "Kept in Django 2.0 for usage by apps supporting Python 2. Shouldn't be used in Django anymore."

File: docs/releases/2.0.txt [hunk 4]
```diff
@@ -205,6 +205,17 @@ Validators
 Backwards incompatible changes in 2.0
 =====================================
 
+Handling of strings vs bytestrings
```
Comment (timgraham) [inline docs/releases/2.0.txt]: maybe: "Support for bytestrings is removed in some places"

File: docs/releases/2.0.txt [hunk 5]
```diff
@@ -205,6 +205,17 @@ Validators
...
+----------------------------------
+
+In older Django releases, having to support native Python 2 strings in addition
```
Comment (timgraham) [inline docs/releases/2.0.txt]: To support native Python 2 strings, older Django versions had to accept both bytestrings and unicode strings. Now that Python 2 support is dropped, bytestrings should only be encountered around input/output boundaries (handling of binary fields or HTTP streams, for example). You might have to update your code to limit bytestring usage to a minimum, as Django no longer accepts bytestrings in certain code paths.

**Comment by claudep** [comment]
Mmmhh, what did I do with that??? It was not committed to master, I should have been tired...

**Comment by claudep** [comment]
Pushed in 52138b1fd08

**Pull request closed without merging**